test/migrate/checks.go

```diff
@@ -0,0 +1,52 @@
+package migrate
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CheckStoredVersions verifies that all status.storedVersions from Knative CRDs are listed in the spec
+// with storage: true. It means the CRDs have been migrated and previous/unused API versions
```

Comment:
This is not the case

You can have N `storedVersions` technically but you can only have a single version in the spec with `storage: true`

It's on us to migrate resources to newer versions and then update the status block removing the older version

Reply:
If your intent is to use this after migration post-install job - then I'd probably name this something more explicit.

maybe something link

```
ExpectSingleStoredVersion
```

Reply:
OK. Hopefully the latest changes address your comments. Sorry it took so long. I didn't notice your comments.